### PR TITLE
Improve log tracing 

### PR DESCRIPTION
**Pull request by Didainius**
It becomes very tricky to track logs when a lot of requests are going in parallel. This PR utilizes new functionality in SDK that helps log correlation. https://github.com/vmware/go-vcloud-director/pull/656

### CODE REVIEW

File: vcd/config_test.go [hunk 1]
```diff
@@ -960,7 +960,9 @@ func getTestVCDFromJson(testConfig TestConfig) (*govcd.VCDClient, error) {
 	}
 	vcdClient := govcd.NewVCDClient(*configUrl, true,
 		govcd.WithSamlAdfs(testConfig.Provider.UseSamlAdfs, testConfig.Provider.CustomAdfsRptId),
-		govcd.WithHttpUserAgent(buildUserAgent("test", testConfig.Provider.SysOrg)))
+		govcd.WithHttpUserAgent(buildUserAgent("test", testConfig.Provider.SysOrg)),
+		govcd.WithVcloudRequestIdFunc(govcd.VcloudRequestIdBuilderFunc),
```
Comment (Didainius) [inline vcd/config_test.go:964]: Client that is initialized for "tests"

File: vcd/config.go [hunk 2]
```diff
@@ -754,6 +754,7 @@ func (c *Config) Client() (*VCDClient, error) {
 			govcd.WithSamlAdfs(c.UseSamlAdfs, c.CustomAdfsRptId),
 			govcd.WithHttpUserAgent(userAgent),
 			govcd.WithIgnoredMetadata(c.IgnoredMetadata),
+			govcd.WithVcloudRequestIdFunc(govcd.VcloudRequestIdBuilderFunc),
```
Comment (Didainius) [inline vcd/config.go:757]: Real client that is used in production code

File: .changes/v3.13.0/1234-notes.md [hunk 3]
```diff
@@ -0,0 +1 @@
+* Improve log traceability by sending `X-VMWARE-VCLOUD-CLIENT-REQUEST-ID` header in requests [GH-1234]
```
[the remark this answers is not in the document]
Comment (Didainius) [reply .changes/v3.13.0/1234-notes.md]: I added a paragraph that should hopefully give more details

**Review by lvirbalas — APPROVED** [review]
Thanks!

**Review by adambarreiro — APPROVED** [review]

**Comment by Didainius** [comment]
Already merged with other PRs

**Pull request closed without merging**